superset-frontend/plugins/legacy-preset-chart-deckgl/src/DeckGLContainer.tsx

```diff
@@ -20,38 +20,41 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ReactNode } from 'react';
 import { isEqual } from 'lodash';
 import { StaticMap } from 'react-map-gl';
-import DeckGL from 'deck.gl';
-import { styled } from '@superset-ui/core';
-import Tooltip from './components/Tooltip';
+import DeckGL, { Layer } from 'deck.gl/typed';
+import { JsonObject, JsonValue, styled } from '@superset-ui/core';
+import Tooltip, { TooltipProps } from './components/Tooltip';
 import 'mapbox-gl/dist/mapbox-gl.css';
+import { Viewport } from './utils/fitViewport';
 
 const TICK = 250; // milliseconds
 
-const propTypes = {
-  viewport: PropTypes.object.isRequired,
-  layers: PropTypes.array.isRequired,
-  setControlValue: PropTypes.func,
-  mapStyle: PropTypes.string,
-  mapboxApiAccessToken: PropTypes.string.isRequired,
-  children: PropTypes.node,
-  bottomMargin: PropTypes.number,
-  width: PropTypes.number.isRequired,
-  height: PropTypes.number.isRequired,
-  onViewportChange: PropTypes.func,
+export type DeckGLContainerProps = {
+  viewport: Viewport;
+  setControlValue?: (control: string, value: JsonValue) => void;
+  mapStyle?: string;
+  mapboxApiAccessToken: string;
+  children?: ReactNode;
+  width: number;
+  height: number;
+  layers: (Layer | (() => Layer))[];
+  onViewportChange?: (viewport: Viewport) => void;
 };
-const defaultProps = {
-  mapStyle: 'light',
-  setControlValue: () => {},
-  children: null,
-  bottomMargin: 0,
+
+export type DeckGLContainerState = {
+  lastUpdate?: number | null;
```

Comment:
Can optional be sufficient here or `null` has a different meaning than `undefined`?

Reply:
It could be undefined because it wasn't defined in the initial state, and only later it was set to a number or null.
It was confusing so now lastUpdate is initialised with null and the type is `lastUpdate: number | null;`. Thanks for catching that!